apps/web/src/contexts/exercise-context.tsx

```diff
@@ -0,0 +1,70 @@
+import { extractStringFromAnyDocument } from '@editor/plugins/text/utils/static-extract-text'
+import { AnyEditorDocument } from '@editor/types/editor-plugins'
+import {
+  isExerciseDocument,
+  isSolutionDocument,
+} from '@editor/types/plugin-type-guards'
+import { createContext, ReactNode, useContext } from 'react'
+
+interface ExerciseContextProps {
+  question: string
+  strategy: string
+  steps: string
+}
+
+interface ExerciseProviderProps {
+  children: ReactNode
+  value: ExerciseContextProps
+}
+
+const ExerciseContext = createContext<ExerciseContextProps | undefined>(
+  undefined
+)
+
+export function ExerciseProvider({ children, value }: ExerciseProviderProps) {
+  return (
+    <ExerciseContext.Provider value={value}>
+      {children}
+    </ExerciseContext.Provider>
+  )
+}
```

Comment:
Isn't this the same as this?
```suggestion
export function ExerciseProvider = ExerciseContext.Provider
```